### PR TITLE
bump go to 1.20

### PR DESCRIPTION
**Pull request by dprotaso**

### CODE REVIEW

File: .github/workflows/reusable-go-build.yaml [hunk 1]
```diff
@@ -25,9 +25,9 @@ jobs:
 
     steps:
       - name: Set up Go
```
Comment (kvmware) [inline .github/workflows/reusable-go-build.yaml:27]: Do we even need to setup go? the runners some with golang installed

**Comment by dprotaso** [comment]
We want to control the go-version and not leave it up to the runner

ie. the default go version is 1.19 until it cuts over April 3rd - https://github.com/actions/runner-images/issues/7276

**Pull request merged**